### PR TITLE
Xpetra: MueLu: require Kokkos

### PR DESCRIPTION
**Pull request by jhux2**

### CODE REVIEW

**Comment by jhux2** [comment]
@trilinos/muelu @trilinos/xpetra

**Review by csiefer2 — APPROVED** [review]

**Review by mayrmt — APPROVED** [review]

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp [hunk 1]
```diff
@@ -82,11 +82,9 @@
 
 // NOTE: We should be checking for KokkosKernels here, but
 // MueLu doesn't have a macro for that
-#ifdef HAVE_MUELU_KOKKOSCORE
 #include "MueLu_LWGraph_kokkos.hpp"
 #include <KokkosGraph_Distance1ColorHandle.hpp>
 #include <KokkosGraph_Distance1Color.hpp>
-#endif
```
Comment (lucbv) [inline packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp:89]: The two "KokkosGraph" headers are coming from Kokkos Kernels which is not a hard dependency, they should be guarded with `#ifdef HAVE_MUELU_KOKKOSKERNELS`.

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp [hunk 2]
```diff
@@ -224,17 +222,11 @@ namespace MueLu {
 ;      TEUCHOS_TEST_FOR_EXCEPTION(A->getRowMap()->getComm()->getSize() != 1, std::invalid_argument,"MIS on more than 1 MPI rank is not supported");
       DoMISNaive(*graph,myColors,numColors);
     }
-#ifdef HAVE_MUELU_KOKKOSCORE
```
Comment (lucbv) [inline packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp:227]: Same as above, use `#ifdef HAVE_MUELU_KOKKOSKERNELS`

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp [hunk 3]
```diff
@@ -337,7 +329,6 @@ GenerateCoarseMap(const Map & fineMap, LO num_c_points, RCP<const Map> & coarseM
 template <class Scalar,class LocalOrdinal, class GlobalOrdinal, class Node>
 void ClassicalMapFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
 DoGraphColoring(const GraphBase & graph, ArrayRCP<LO> & myColors_out, LO & numColors) const {
-#ifdef HAVE_MUELU_KOKKOSCORE
```
Comment (lucbv) [inline packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp:340]: Same as above

File: packages/xpetra/cmake/Xpetra_config.hpp.in [hunk 4]
```diff
@@ -24,8 +24,6 @@
 
 #cmakedefine HAVE_XPETRA_KOKKOSCONTAINERS
 
-#cmakedefine HAVE_XPETRA_KOKKOSCORE
```
Comment (lucbv) [inline packages/xpetra/cmake/Xpetra_config.hpp.in:27]: We might want to define `#cmakedefine HAVE_XPETRA_KOKKOSKERNELS`?

**Review by lucbv — CHANGES_REQUESTED** [review]
Some code paths need to be guarded with `#ifdef HAVE_MUELU_KOKKOSKERNELS`, they were not guarded properly until know...

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp (hunk 1 above)
Comment (jhux2) [reply packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp:89]: addressed

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp (hunk 2 above)
Comment (jhux2) [reply packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp:227]: addressed

File: packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp (hunk 3 above)
Comment (jhux2) [reply packages/muelu/src/Transfers/Classical/MueLu_ClassicalMapFactory_def.hpp:340]: addressed

File: packages/xpetra/cmake/Xpetra_config.hpp.in (hunk 4 above)
Comment (jhux2) [reply packages/xpetra/cmake/Xpetra_config.hpp.in:27]: addressed

**Review by cgcgcg — APPROVED** [review]

**Review by lucbv — APPROVED** [review]
lgtm, thanks for the updates : )

**Comment by jhux2** [comment]
@trilinos/framework @william76 @ZUUL42  Even though the autotester reports failures, I see no failures on the dashboard.  I'm not sure what the problem is.

**Comment by jhux2** [comment]
> @trilinos/framework @william76 @ZUUL42 Even though the autotester reports failures, I see no failures on the dashboard. I'm not sure what the problem is.

Oh, this doesn't look good:
```
================================================================================
git remote exists, removing it

+----------------------------------------------------------+
|              An error occurred during merge              |
|                                                          |
|                   2021-09-10 05:34:02                    |
+----------------------------------------------------------+
Build step 'Execute shell' marked build as failure
Archiving artifacts
[Checks API] No suitable checks publisher found.
Finished: FAILURE
```

**Comment by jhux2** [comment]
maybe the AT will work this time

**Comment by jhux2** [comment]
Blocked by #9686.

**Pull request merged**